src/tests/simulation_scale/src/ctl_ext.rs

```diff
@@ -68,31 +79,33 @@ pub mod predicate {
         Box::new(p)
     }
 
-    /// The fragment is able to be scaled. Used for locating random fragment.
-    pub fn can_scale() -> BoxedPredicate {
+    /// The fragment is able to be rescheduled. Used for locating random fragment.
+    pub fn can_reschedule() -> BoxedPredicate {
         let p = |f: &ProstFragment| {
-            // TODO: singleton fragments are also able to be migrated, may add a `can_migrate`
-            // predicate.
-            let distributed = f.distribution_type() == FragmentDistributionType::Hash;
+            // FIXME: we should already support reschedule source fragment, but there might be a bug
+            // of nexmark generator recovery.
+            let has_source = identity_contains("StreamSource")(f);
```

Comment:
If we reschedule the nexmark source executors, it seems the result will possibly be wrong. 🤔 cc @shanicky

Reply:
I'll take a look after this PR first.